File: src/tone.cxx
#include "tone.hxx"

namespace Moon {
  Tone::Tone() {
    hue = 1.0;
    saturation = 1.0;
    lightness = 1.0;
  };

  Tone::Tone(GLfloat hue, GLfloat saturation, GLfloat lightness) {
    this->hue = hue;
    this->saturation = saturation;
    this->lightness = lightness;
  };
};

File: core/tone.rb
#
# moon/core/tone.rb
#
class Moon::Tone

  alias :h :hue
  alias :h= :hue=
  alias :s :saturation
  alias :s= :saturation=
  alias :l :lightness
  alias :l= :lightness=

  def dup
    Tone.new h, s, l
  end

  def to_a
    return h, s, l
  end

  def to_h
    {
      h: h,
      s: s,
      l: l
    }
  end

end
